modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryServiceImpl.java

```diff
@@ -112,6 +112,24 @@ public ListTypeEntry getListTypeEntry(long listTypeEntryId)
 		return listTypeEntryLocalService.getListTypeEntry(listTypeEntryId);
 	}
 
+	@Override
+	public ListTypeEntry getListTypeEntryByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		ListTypeEntry listTypeEntry =
+			listTypeEntryLocalService.fetchListTypeEntryByExternalReferenceCode(
+				externalReferenceCode, companyId);
+
+		_listTypeDefinitionModelResourcePermission.check(
+			getPermissionChecker(), listTypeEntry.getListTypeDefinitionId(),
+			ActionKeys.VIEW);
+
+		return listTypeEntryLocalService.
```

Comment:
You don't need to get the object again here, just return the **listTypeEntry** you got before.